java/android/nnstreamer/src/main/jni/Android.mk

```diff
@@ -116,79 +173,75 @@ include $(LOCAL_PATH)/Android-nnstreamer.mk
 # external libs and sub-plugins
 #------------------------------------------------------
 ifeq ($(_ENABLE_TF_LITE),true)
-NNS_API_FLAGS += -DENABLE_TENSORFLOW_LITE=1
-NNS_SUBPLUGINS += tensorflow-lite-subplugin
-
 include $(LOCAL_PATH)/Android-tensorflow-lite.mk
+NNS_SUBPLUGINS += tensorflow-lite-subplugin
 endif
 
 ifeq ($(ENABLE_SNAP),true)
-NNS_API_FLAGS += -DENABLE_SNAP=1
-NNS_SUBPLUGINS += snap-subplugin
-
 include $(LOCAL_PATH)/Android-snap.mk
+NNS_SUBPLUGINS += snap-subplugin
 endif
 
 ifeq ($(ENABLE_NNFW),true)
-NNS_API_FLAGS += -DENABLE_NNFW_RUNTIME=1
-NNS_SUBPLUGINS += nnfw-subplugin
-
 include $(LOCAL_PATH)/Android-nnfw.mk
+NNS_SUBPLUGINS += nnfw-subplugin
 endif
 
 ifeq ($(_ENABLE_SNPE),true)
-NNS_API_FLAGS += -DENABLE_SNPE=1
-NNS_SUBPLUGINS += snpe-subplugin
-
 include $(LOCAL_PATH)/Android-snpe.mk
+NNS_SUBPLUGINS += snpe-subplugin
 endif
 
 ifeq ($(ENABLE_QNN),true)
-NNS_API_FLAGS += -DENABLE_QNN=1
-NNS_SUBPLUGINS += qnn-subplugin
-
 include $(LOCAL_PATH)/Android-qnn.mk
+NNS_SUBPLUGINS += qnn-subplugin
 endif
 
 ifeq ($(ENABLE_PYTORCH),true)
-NNS_API_FLAGS += -DENABLE_PYTORCH=1
-NNS_SUBPLUGINS += pytorch-subplugin
-
 include $(LOCAL_PATH)/Android-pytorch.mk
+NNS_SUBPLUGINS += pytorch-subplugin
 endif
 
 ifeq ($(ENABLE_MXNET), true)
-NNS_API_FLAGS += -DENABLE_MXNET=1
-NNS_SUBPLUGINS += mxnet-subplugin
-
 include $(LOCAL_PATH)/Android-mxnet.mk
+NNS_SUBPLUGINS += mxnet-subplugin
 endif
 
 ifeq ($(ENABLE_LLAMA2C), true)
-NNS_API_FLAGS += -DENABLE_LLAMA2C=1
-NNS_SUBPLUGINS += llama2c-subplugin
-
 include $(LOCAL_PATH)/Android-llama2c.mk
+NNS_SUBPLUGINS += llama2c-subplugin
 endif
 
 ifeq ($(ENABLE_LLAMACPP), true)
-NNS_API_FLAGS += -DENABLE_LLAMACPP=1
-NNS_SUBPLUGINS += llamacpp-subplugin
-
 include $(LOCAL_PATH)/Android-llamacpp.mk
+NNS_SUBPLUGINS += llamacpp-subplugin
 endif
 
-ifneq ($(NNSTREAMER_API_OPTION),single)
 ifeq ($(ENABLE_FLATBUF),true)
 include $(LOCAL_PATH)/Android-flatbuf.mk
-NNS_API_FLAGS += -DENABLE_FLATBUF=1
 NNS_SUBPLUGINS += flatbuffers-subplugin
 endif
-endif
 
 # Remove any duplicates.
 NNS_SUBPLUGINS := $(sort $(NNS_SUBPLUGINS))
 
+#------------------------------------------------------
+# nnstreamer for android
+#------------------------------------------------------
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := nnstreamer_android
+
+LOCAL_SRC_FILES := nnstreamer-native-init.c
+LOCAL_C_INCLUDES := $(NNSTREAMER_INCLUDES) $(NNSTREAMER_CAPI_INCLUDES) $(GST_HEADERS_COMMON)
+LOCAL_CFLAGS := -O3 -fPIC $(NNS_API_FLAGS)
+LOCAL_STATIC_LIBRARIES := nnstreamer $(NNS_SUBPLUGINS)
+
+LOCAL_EXPORT_LDLIBS := $(NNS_API_FLAGS)
```

Comment:
`LOCAL_EXPORT_LDLIBS` requires library names not compile args. Can be omitted?

Reply:
thanks, updated :)